src/BeamAdapter/component/mapping/AdaptiveBeamMapping.inl

```diff
@@ -237,21 +250,42 @@ void AdaptiveBeamMapping< TIn, TOut>::apply(const MechanicalParams* mparams, Dat
     }
 
     AdvancedTimer::stepBegin("computeNewInterpolation");
-    for(unsigned int i=0; i<m_pointBeamDistribution.size(); i++)
-    {
-        PosPointDefinition pointBeamDistribution = m_pointBeamDistribution[i];
-        Vec<3, InReal> pos;
-        const Vec3 localPos(0.,pointBeamDistribution.baryPoint[1],pointBeamDistribution.baryPoint[2]);
-        l_adaptativebeamInterpolation->interpolatePointUsingSpline(pointBeamDistribution.beamId, pointBeamDistribution.baryPoint[0], localPos, in, pos, false, xtest);
 
-        if(m_isSubMapping)
+    // effective computation for each element in pointBeamDistribution
+    auto apply_impl = [&](const auto& range)
+    {
+        auto elementID = std::distance(m_pointBeamDistribution.begin(), range.start);
+        for (auto it = range.start; it != range.end; ++it, ++elementID)
         {
-            if(m_idPointSubMap.size()>0)
-                out[m_idPointSubMap[i]] = pos;
+            const auto& pointBeamDistribution = m_pointBeamDistribution[elementID];
+
+            Vec<3, InReal> pos(sofa::type::NOINIT);
+            const Vec3 localPos(0., pointBeamDistribution.baryPoint[1], pointBeamDistribution.baryPoint[2]);
+            l_adaptativebeamInterpolation->interpolatePointUsingSpline(pointBeamDistribution.beamId, pointBeamDistribution.baryPoint[0], localPos, in, pos, false, xtest);
+
+            if (m_isSubMapping)
+            {
+                if (m_idPointSubMap.size() > 0)
```

Comment:
easier to read:
```suggestion
                if (!m_idPointSubMap.empty())
```

Reply:
https://clang.llvm.org/extra/clang-tidy/checks/readability/container-size-empty.html